static/app/components/lazyLoad.tsx

```diff
@@ -12,137 +12,111 @@ type PromisedImport<C> = Promise<{default: C}>;
 
 type ComponentType = React.ComponentType<any>;
 
-type Props<C extends ComponentType> = Omit<
-  React.ComponentProps<C>,
-  'hideBusy' | 'hideError' | 'component' | 'route'
-> & {
+type Props<C extends ComponentType> = Omit<React.ComponentProps<C>, 'route'> & {
   /**
-   * Function that returns a promise of a React.Component
+   * Accepts a function to trigger the import resolution of the component.
    */
   component?: () => PromisedImport<C>;
-  hideBusy?: boolean;
-  hideError?: boolean;
   /**
-   * Also accepts a route object from react-router that has a `componentPromise` property
+   * Accepts a route object from react-router that has a `componentPromise` property
    */
   route?: {componentPromise: () => PromisedImport<C>};
 };
 
-type State<C extends ComponentType> = {
-  LazyComponent: C | null;
-  error: any | null;
-};
-
-class LazyLoad<C extends ComponentType> extends Component<Props<C>, State<C>> {
-  state: State<C> = {
-    LazyComponent: null,
-    error: null,
-  };
-
-  componentDidMount() {
-    this.fetchComponent();
-  }
-
-  UNSAFE_componentWillReceiveProps(nextProps: Props<C>) {
-    // No need to refetch when component does not change
-    if (nextProps.component && nextProps.component === this.props.component) {
+/**
+ * LazyLoad is used to dynamically load codesplit components via a `import`
+ * call. Typically this component is used as part of the routing tree, though
+ * it does have a standalone mode.
+ *
+ * Route tree usage:
+ *   <Route
+ *     path="somePath"
+ *     component={LazyLoad}
+ *     componentPromise={() => import('./somePathView')}
```

Comment:
fwiw I will be removing this feature in the near future. Since react-router 6 does not support arbitrary route props.